### PR TITLE
Add non-aws S3 support

### PR DESCRIPTION
**Pull request by qmnonic**
Allow users to specify the endpoint_url in the config file.

### CODE REVIEW

**Comment by koszti** [comment]
hey @qmnonic , this is great feature, but it breaks the default setup when `aws_endpoint_url` is not defined:

`File "/Users/peter.kosztolanyi/workspaces/transferwise/pipelinewise-tap-s3-csv/venv/lib/python3.7/site-packages/tap_s3_csv/s3.py", line 160, in get_input_files_for_table
    for s3_object in list_files_in_bucket(bucket, prefix, aws_endpoint_url=config['aws_endpoint_url']):
KeyError: 'aws_endpoint_url'`

Can we make `aws_endpoint_url` optional in config.json?

File: tap_s3_csv/s3.py [hunk 1]
```diff
@@ -225,7 +232,14 @@ def list_files_in_bucket(bucket, search_prefix=None):
 @retry_pattern()
 def get_file_handle(config, s3_path):
     bucket = config['bucket']
-    s3_client = boto3.resource('s3')
+    aws_endpoint_url = config['aws_endpoint_url']
```
Comment (koszti) [inline tap_s3_csv/s3.py]: it gives `KeyError: 'aws_endpoint_url'` when it's missing from `config.json`

File: tap_s3_csv/s3.py [hunk 2]
```diff
@@ -99,6 +99,7 @@ def merge_dicts(first, second):
 
 def sample_file(config, table_spec, s3_path, sample_rate):
     file_handle = get_file_handle(config, s3_path)
+    # _raw_stream seems like the wrong way to access this..
```
Comment (koszti) [inline tap_s3_csv/s3.py]: can you please give a little more details and suggestions about it?

Comment (qmnonic) [reply tap_s3_csv/s3.py]: it appears to be a private property, which means there is probably a better way to iterate over the file.  i don't have a suggestion.

File: tap_s3_csv/s3.py (hunk 1 above)
Comment (qmnonic) [reply tap_s3_csv/s3.py]: adjusted to use config.get('aws_endpoint_url'), which should return None, if it doesn't exist.

File: tap_s3_csv/s3.py [hunk 3]
```diff
@@ -156,7 +157,7 @@ def get_input_files_for_table(config, table_spec, modified_since=None):
     matched_files_count = 0
     unmatched_files_count = 0
     max_files_before_log = 30000
-    for s3_object in list_files_in_bucket(bucket, prefix):
+    for s3_object in list_files_in_bucket(bucket, prefix, aws_endpoint_url=config['aws_endpoint_url']):
```
Comment (koszti) [inline tap_s3_csv/s3.py]: `KeyError: 'aws_endpoint_url'` here as well when not in `config.json`

can we please use `.get` instead?

Comment (qmnonic) [reply tap_s3_csv/s3.py]: updated.

**Comment by koszti** [comment]
thank you, approved. Please go ahead and merge and will publish 1.0.5 to pypi.

**Pull request merged**